### PR TITLE
ddl/table_split_test.go: migrate test-infra to testify

### PR DESCRIPTION
**Pull request by wangggong**
Signed-off-by: wangggong <793160615@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29115

Problem Summary:

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by wangggong** [comment]
/cc @tisonkun 

PTAL

File: ddl/table_split_test.go [hunk 1]
```diff
@@ -56,32 +57,32 @@ func (s *testDDLTableSplitSuite) TestTableSplit(c *C) {
 	defer dom.Close()
 	atomic.StoreUint32(&ddl.EnableSplitTableRegion, 0)
 	infoSchema := dom.InfoSchema()
-	c.Assert(infoSchema, NotNil)
-	t, err := infoSchema.TableByName(model.NewCIStr("mysql"), model.NewCIStr("tidb"))
-	c.Assert(err, IsNil)
-	checkRegionStartWithTableID(c, t.Meta().ID, store.(kvStore))
+	require.NotNil(t, infoSchema, nil)
```
Comment (hawkingrei) [inline ddl/table_split_test.go:60]: this nil can be removed.

Comment (tisonkun) [reply ddl/table_split_test.go:60]: Agree. Let me push a trivial patch for this.
```suggestion
	require.NotNil(t, infoSchema)
```

File: ddl/table_split_test.go [hunk 2]
```diff
@@ -56,32 +57,32 @@ func (s *testDDLTableSplitSuite) TestTableSplit(c *C) {
...
+	checkRegionStartWithTableID(t, tbl.Meta().ID, store.(kvStore))
 
-	t, err = infoSchema.TableByName(model.NewCIStr("test"), model.NewCIStr("t_part"))
-	c.Assert(err, IsNil)
-	pi := t.Meta().GetPartitionInfo()
-	c.Assert(pi, NotNil)
+	tbl, err = infoSchema.TableByName(model.NewCIStr("test"), model.NewCIStr("t_part"))
+	require.NoError(t, err)
+	pi := tbl.Meta().GetPartitionInfo()
+	require.NotNil(t, pi, nil)
```
Comment (tisonkun) [inline ddl/table_split_test.go:68]:
```suggestion
	require.NotNil(t, pi)
```

**Review by tisonkun — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**